test/systemtests/kubeadm_test.go

```diff
@@ -0,0 +1,981 @@
+	logrus.Debugf("ExecBG: Running command -- %s", cmd)
+	k8sMaster.tbnode.RunCommandBackground(cmd)
+}
+
+/*
+* podExec function is used to run a command typically multiple commands
+* with pipes and redirections within the pod rather than the node
+ */
+func (k *kubePod) podExec(podName, args string, ns ...string) (string, error) {
+	// NOTE:
+	// Remotessh library wraps this command as follows:
+	// ssh <hostip> <port> <env-var> bash -lc '<cmd>'
+	//
+	// Backticks and quotes here ensure that the command
+	// is properly wrapped to execute on pod rather than on the node
+	podCmd := `sh -c '\''` + args + `'\''`
+	return k.exec(podName, podCmd, ns...)
+}
+
+/*
+* podExecBG function is used to run a background command
+* within the pod rather than the node
+ */
+func (k *kubePod) podExecBG(podName, args string, ns ...string) error {
+	namespace := "default"
+	if len(ns) != 0 {
+		namespace = ns[0]
+	}
+
+	// NOTE:
+	// Remotessh library wraps this command as follows:
+	// ssh <hostip> <port> <env-var> bash -lc '<cmd>'
+	//
+	//  - Backticks and quotes here ensure that the command
+	//    is properly wrapped to execute on pod rather than on the node
+	//  - nohup along with & ensures that the process continues to live
+	//    after the shell is terminated
+	// Since the command is to run in BG in the pod and not on the node,
+	// RunCommandBackground is not required here
+
+	podCmd := `kubectl -n ` + namespace + ` exec ` + podName + ` -- nohup sh -c '\''` + args + ` &'\''`
+	logrus.Debugf("Pod Exec BG: Running command -- %s", podCmd)
+	return k8sMaster.tbnode.RunCommand(podCmd)
+}
+
+func (k *kubePod) start(c *container) error {
+	//Kubernetes does not support start/stop
+	return nil
+}
+
+func (k *kubePod) stop(c *container) error {
+	//Kubernetes does not support start/stop
+	return nil
+}
+
+func (k *kubePod) rm(c *container) error {
+	logrus.Infof("Removing Pod: %s on %s", c.containerID, c.node.Name())
+	k8sMaster.tbnode.RunCommand(fmt.Sprintf("kubectl delete pod %s", c.name))
+	for i := 0; i < 80; i++ {
+		out, _ := k8sMaster.tbnode.RunCommandWithOutput(fmt.Sprintf("kubectl get pod %s", c.containerID))
+		if strings.Contains(out, "not found") {
+			return nil
+		}
+		time.Sleep(5 * time.Second)
+	}
+	return fmt.Errorf("Error Termininating pod %s on node %s", c.name, c.node.Name())
+}
+
+func (k *kubePod) startListener(c *container, port int, protocol string) error {
+	var protoStr string
+
+	if protocol == "udp" {
+		protoStr = "-u"
+	}
+
+	k.execBG(c.containerID, fmt.Sprintf("nc -lk %s -p %v -e /bin/true", protoStr, port))
+	return nil
+
+}
+
+func (k *kubePod) startIperfServer(c *container) error {
+
+	k.execBG(c.containerID, fmt.Sprintf("iperf -s"))
+	return nil
+}
+
+func (k *kubePod) startIperfClient(c *container, ip, limit string, isErr bool) error {
+
+	var (
+		bwLimit int64
+		bwInt64 int64
+	)
+	bw, err := k.exec(c.containerID, fmt.Sprintf("iperf -c %s ", ip))
+	logrus.Infof("starting iperf client on : %v for server ip: %s", c, ip)
```

Comment:
NOTE: Haven't changed these functions at all from legacy k8s mode. So, I have to look at these functions and simplify/refactor them as needed.